azext_iot/operations/events3/_events.py

```diff
@@ -125,7 +127,14 @@ def _get_conn_props():
     def _output_msg_kpi(msg):
         # TODO: Determine if amqp filters can support boolean operators for multiple conditions
         origin = str(msg.annotations.get(b'iothub-connection-device-id'), 'utf8')
-        if device_id and origin != device_id:
+        if device_id and device_id != origin:
+            if '*' in device_id or '?' in device_id:
+                regex = re.escape(device_id).replace("\\*", ".*").replace('\\?', ".") + "$"
+                if not re.match(regex, origin):
+                    return
+            else:
+                return
+        if devices and origin not in devices:
```

Comment:
Use dict.get() instead of iteration through keys.

Reply:
Can you elaborate a bit? We just want to check if that value is in the "devices" dictionary (x in y), we don't need the value (even though they're only set to True when we create the dictionary).

In my testing, the `key in dict` lookup was more performant - though I understand `dict.get()` is safer if we're using the resultant value since we can specify a default / no-exist value.

Reply:
The thought was, iterating through a list of keys (particularly when sufficient large) is slower than a safe (due to default value) direct look up.

Reply:
Just wrote a small test script for this, it does looks like 'in' is faster like you mentioned. This is probably due to bytecode optimization and not needing to lookup a member attribute get, and executing method (expensive).

Kudos.

Reply:
Side note/clarification, dict[...] is the expected performant lookup.

Reply:
I see...from my understanding, as long as the right-hand operand is a dictionary, python does a key lookup with the "in" keyword, as opposed to iterating through the keys.